File: MobiliteInternationalBackend/src/main/java/com/example/chadi/Entity/Candidature.java
package com.example.chadi.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Candidature implements Serializable {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private Integer idCandidature;
    private String NomComplet;
    private String Identifiant;
    private String email;
    @Enumerated(EnumType.STRING)
    private  Genre genre;
    private LocalDate DateNaissance;
    private String Classe;
    private double Moy3;
    private double Moy2;
    private double Moy1;
    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
    private double score;
    @Enumerated(EnumType.STRING)
    private  Niveau Francais;
    @Enumerated(EnumType.STRING)
    private  Niveau Anglais;
    @Enumerated(EnumType.STRING)
    private  Option specialite;
    @ManyToOne
    @JsonIgnore
    User userCandidature;
    @ManyToOne
    @JsonIgnore
    Offre offre;



}

File: MobiliteInternationalFrontend/src/app/core/model/Candidature.ts
import { offre } from "./offre";
export enum Genre {
    Masculin,
    Féminin
    }
    export enum Niveau {
        A1,A2,B1,B2,C1,C2,Null
        }
        export enum option {
            ERPBI,arcTIC,SAE,SIM,SLEAM,NIDS,TWIN,InFini,DataScience,SE,IoSyS,Wins,OGI,Mecatrononique,StructureOuvrage,PontsEtchaussées,RessourcesNaturellesEtétudesEnergétique,Autre
        }
export class Candidature {
    constructor(
      public idCandidature: number,
      public nomComplet: string,
      public identifiant: string,
      public email: string,
      public gerne: Genre,
      public dateNaissance: Date,
      public classe: string,
      public moy3: number,
      public moy2: number,
      public moy1: number,
      public score: number,
      public francais: Niveau,
      public anglais: Niveau,
      public option: option,

      public offre: offre, 
          //public userTicket: User
    ) {}
  }